war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -1658,9 +1658,25 @@ function expandTextArea(button,id) {
 // by using the contents fetched from the given URL.
 function refreshPart(id,url) {
     var intervalID = null;
+    var refreshInterval = 5000; //default is 5s
+
     var f = function() {
         if(isPageVisible()) {
             new Ajax.Request(url, {
+                onCreate: function() {
+                    requestStartTime = new Date().getTime();
+                    console.log("onCreate send of id:"+id+", url:"+url+", start time:"+requestStartTime);
+                },
+                onComplete: function() {
+                    var requestDuration = new Date().getTime() - requestStartTime;
+                    if(requestDuration > refreshInterval) {
+                        //server or network has issues, slowing ajax calls. Will be reset by a page refresh (F5)
+                        refreshInterval = 2 * refreshInterval;
+                        window.clearInterval(intervalID);
+                        intervalID = window.setInterval(f, refreshInterval);
+                    }
```

Comment:
need to add some logic so that the refresh interval can decrease if response time increases

Reply:
I don't think so, because a refresh or a page change reset to 5s so I think it's ok and less complex like that. If Jenkins is not responding people are going to hit F5.

Reply:
People reloading the page seems like it would make this approach ineffective - possibly we could store the "delay" variable in a cookie.  This should also make it available across any page that makes ajax requests.